tfhe/src/integer/mod.rs

```diff
@@ -76,6 +76,14 @@ pub use client_key::{ClientKey, CrtClientKey, RadixClientKey};
 pub use public_key::{CompressedCompactPublicKey, CompressedPublicKey, PublicKey};
 pub use server_key::{CheckError, CompressedServerKey, ServerKey};
 
+/// Enum to indicate which kind of computations the [`ServerKey`] will be performing, this changes
+/// the parameterization of the key to manage carries in the Radix case.
+#[derive(Clone, Copy, Debug)]
+pub enum IntegerKeyKind {
+    Radix,
+    CRT,
+}
```

Comment:
Would it make sense to later have to disctinct ServerKey struct instead ? 
As with this enum a use could potentially using function that do not match the kind

Reply:
I was wondering about that too, at the moment the design is this way, and that's why I asked if we imagined having a key able to do both

Reply:
We'll see how it should evolve

Reply:
Ok, will open a follow up issue

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/321